Use the canonical approach and propagate the type

garronej opened the pull request:
Just an idea

kathari00 commented:
@luca-peruzzo this approach has the advantage that it is typed better as messages are being suggested when using it. What do you think about that?

garronej commented:
Hi @kathari00,

I’ve updated this PR to use Keycloakify 11.

Since this is actually the first time I touch an Angular codebase, there might be small mistakes, but I’m confident in the overall validity of the proposed changes.

I’ve removed the `i18n-theme-defined-messages.provider.ts` file, as its functionality is now handled by the `src/keycloak-theme/login/i18n.ts` file (see the comments where I’ve included relevant links).

Things that you might want to know to tackle on your end:  

- The script that initializes the template has been updated to align with the latest version of Keycloak:  
  [Template Initialization Script](https://github.com/keycloakify/keycloakify/blob/main/src/login/Template.useInitialize.ts)
  
- There’s also an important import map, but it’s provided by the FTL template, so you shouldn’t need to worry about it:  
  [Import Map Reference](https://github.com/keycloakify/keycloakify/blob/4f1141510787dd9f777663d9c2de2b28d97f81bb/src/bin/keycloakify/generateFtl/kcContextDeclarationTemplate.ftl#L90-L100)

  (Since import maps must be present at the head or loaded via non-ESM JavaScript, it’s too late for me to set them in React, so I’ve had to handle it server-side 😞 .)

Let me know if you have any questions or concerns.

Best,
